plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.page.ts

```diff
@@ -606,6 +601,11 @@ class ProductCollectionPage {
 		return this.page.getByRole( 'heading' );
 	}
 
+	async getProductNames() {
+		const products = this.page.locator( '.wp-block-post-title' );
+		return products.allTextContents();
```

Comment:
We need to be careful when getting text content this way as it's inherently flaky. To address this, we should assert the number of elements that we expect to be rendered before getting their content. If we don't do it, we will only get text content from the elements that were rendered at the time this method was called.

Reply:
Good tip! In this case, the products are server-side-rendered so I expect all of them to be visible from the very beginning but I added assertion anyway 👍